### PR TITLE
Remove CA1307 suppression's (final part) 

### PR DESCRIPTION
**Pull request by martinrrm**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1977

Regression? Last working version:

## Description

Continuation of removing CA1307 suppressions 

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception - The only changes that I see that can have tests are the `string.CompareTo` to `string,Compare` but I don't see a change in behavior since they are strings
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VsCredentialProviderImporter.cs [hunk 1]
```diff
@@ -76,7 +76,7 @@ public IReadOnlyCollection<ICredentialProvider> GetProviders()
             TryImportCredentialProviders(results, ImportedProviders);
 
             // Ensure imported providers ordering is deterministic
-            results.Sort((a, b) => a.GetType().FullName.CompareTo(b.GetType().FullName));
+            results.Sort((a, b) => string.Compare(a.GetType().FullName, b.GetType().FullName, StringComparison.Ordinal));
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VsCredentialProviderImporter.cs:79]: style: Don't people use `StringComparer.Ordinal.Compare`?

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VsCredentialProviderImporter.cs:79]: This applies in a few more places should you choose to change. 
Up to you.

**Review by nkolev92 — APPROVED** [review]

**Review by erdembayar — APPROVED** [review]

**Review by zivkan — APPROVED** [review]

**Comment by dominoFire** [comment]
There's a CA1307 suppression in Xplat project: https://github.com/NuGet/NuGet.Client/blob/2495e1c473d2afa94eccaff4323473d0481dbf81/src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj#L6

Those changes were introduced here to unblock source-build work https://github.com/NuGet/NuGet.Client/pull/4404/files#diff-9155a67886383a6a51c5bec24022e5ec0722ce98a1f54fe2f52bc120ce6f0ec7

Some suppressions applies to NuGet.CommandLine.XPlat code; other applies to Microsoft.Extensions.CommandLineUtils, which will be addressed once System.CommandLine migration is done.

Let's follow up on this later https://github.com/NuGet/NuGet.Client/pull/4943

**Review by dominoFire — APPROVED** [review]
Thank you! Let's follow up in https://github.com/NuGet/Client.Engineering/issues/2035

**Pull request merged**